src/benchmarks/micro/libraries/System.Text.Json/Node/ParseThenWrite.cs

```diff
@@ -0,0 +1,89 @@
+﻿using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text.Json.Document.Tests;
+using System.Text.Json.Nodes;
+using System.Text.Json.Tests;
+
+namespace System.Text.Json.Node.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.JSON)]
+    public class Perf_ParseThenWrite
+    {
+        public enum TestCaseType
+        {
+            HelloWorld,
+            DeepTree,
+            BroadTree,
+            LotsOfNumbers,
+            LotsOfStrings,
+            Json400B,
+            Json4KB,
+            Json400KB
+        }
+
+        private byte[] _dataUtf8;
+        private Utf8JsonWriter _writer;
+
+        [ParamsAllValues]
+        public TestCaseType TestCase;
+
+        [Params(true, false)]
+        public bool IsDataIndented;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            string jsonString = JsonStrings.ResourceManager.GetString(TestCase.ToString());
+
+            if (!IsDataIndented)
+            {
+                _dataUtf8 = DocumentHelpers.RemoveFormatting(jsonString);
+            }
+            else
+            {
+                _dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+            }
```

Comment:
```suggestion
            _dataUtf8 = IsDataIndented ?
                Encoding.UTF8.GetBytes(jsonString) :
                DocumentHelpers.RemoveFormatting(jsonString);
```
?

Reply:
> Initialization code is copypasta'd from the equivalent

FWIW Setup and cleanup code can be always refactored and improved (duplicated code moved to helper type), as changing it typically does not affect the benchmark itself